src/System.Runtime/tests/System/Int16.cs

```diff
@@ -38,255 +36,237 @@ public static void TestMinValue()
-
-        numberFormat.NegativeSign = "xx"; // setting it to trash to make sure it doesn't show up
-        numberFormat.NumberGroupSeparator = "*";
-        numberFormat.NumberNegativePattern = 0;
-        short i3 = -2468;
-        Assert.Equal("(2*468.00)", i3.ToString("N", numberFormat));
+        short i = 123;
+        Assert.Throws<FormatException>(() => i.ToString("Y")); // Invalid format
+        Assert.Throws<FormatException>(() => i.ToString("Y", null)); // Invalid format
     }
 
-    public static IEnumerable<object[]> ParseValidData()
+    public static IEnumerable<object[]> Parse_Valid_TestData()
     {
-        NumberFormatInfo defaultFormat = null;
+        NumberFormatInfo nullFormat = null;
         NumberStyles defaultStyle = NumberStyles.Integer;
-        var emptyNfi = new NumberFormatInfo();
+        NumberFormatInfo emptyFormat = new NumberFormatInfo();
 
-        var testNfi = new NumberFormatInfo();
-        testNfi.CurrencySymbol = "$";
+        NumberFormatInfo customFormat = new NumberFormatInfo();
+        customFormat.CurrencySymbol = "$";
 
-        yield return new object[] { "-32768", defaultStyle, defaultFormat, (short)-32768 };
-        yield return new object[] { "-123", defaultStyle, defaultFormat, (short)-123 };
-        yield return new object[] { "0", defaultStyle, defaultFormat, (short)0 };
-        yield return new object[] { "123", defaultStyle, defaultFormat, (short)123 };
-        yield return new object[] { "  123  ", defaultStyle, defaultFormat, (short)123 };
-        yield return new object[] { "32767", defaultStyle, defaultFormat, (short)32767 };
+        yield return new object[] { "-32768", defaultStyle, nullFormat, (short)-32768 };
+        yield return new object[] { "-123", defaultStyle, nullFormat, (short)-123 };
+        yield return new object[] { "0", defaultStyle, nullFormat, (short)0 };
+        yield return new object[] { "123", defaultStyle, nullFormat, (short)123 };
+        yield return new object[] { "  123  ", defaultStyle, nullFormat, (short)123 };
+        yield return new object[] { "32767", defaultStyle, nullFormat, (short)32767 };
 
-        yield return new object[] { "123", NumberStyles.HexNumber, defaultFormat, (short )0x123 };
-        yield return new object[] { "abc", NumberStyles.HexNumber, defaultFormat, (short)0xabc };
-        yield return new object[] { "1000", NumberStyles.AllowThousands, defaultFormat, (short)1000 };
-        yield return new object[] { "(123)", NumberStyles.AllowParentheses, defaultFormat, (short)-123 }; // Parentheses = negative
+        yield return new object[] { "123", NumberStyles.HexNumber, nullFormat, (short)0x123 };
+        yield return new object[] { "abc", NumberStyles.HexNumber, nullFormat, (short)0xabc };
+        yield return new object[] { "1000", NumberStyles.AllowThousands, nullFormat, (short)1000 };
+        yield return new object[] { "(123)", NumberStyles.AllowParentheses, nullFormat, (short)-123 }; // Parentheses = negative
```

Comment:
I wanted to make it clear that this was the IFormatProvider parameter, but I can undo this if you want